CodeGeneration/Sources/SyntaxSupport/Child.swift

```diff
@@ -87,8 +87,27 @@ public class Child {
   /// This is used to e.g. describe the child if all of its tokens are missing in the source file.
   public let nameForDiagnostics: String?
 
-  /// A doc comment describing the child.
-  public let documentation: SwiftSyntax.Trivia
+  /// A summary of a doc comment describing the child. Full docc comment for the
+  /// child is available in ``Child/documentation``, and includes detailed list
+  /// of possible choices for the child if it's a token kind.
+  public let documentationSummary: SwiftSyntax.Trivia
+
+  /// A docc comment describing the child, including the trivia provided when
+  /// initializing the Child, and the list of possible token choices inferred automatically.
+  public var documentation: SwiftSyntax.Trivia {
+    if case .token(let choices, _, _) = kind {
+      var tokenKindsComment = choices.count == 1 ?
+        "For syntax trees generated by the parser, this is guaranteed to be the following kind:" :
```

Comment:
I _think_ we could shave off the extra line and make the pluralization easier, but that's probably fine.